### ISSUE TITLE
SharePoint Admin Api Access page broken

### ISSUE THREAD

**Issue by ntdong1889** [SharePoint/sp-dev-docs#7719]
### Target SharePoint environment

SharePoint Online

### What SharePoint development model, framework, SDK or API is this about?

💥 SharePoint Framework

### Developer environment

Windows

### What browser(s) / client(s) have you tested

- [X] 💥 Internet Explorer
- [X] 💥 Microsoft Edge
- [X] 💥 Google Chrome
- [X] 💥 FireFox
- [X] 💥 Safari
- [ ] mobile (iOS/iPadOS)
- [ ] mobile (Android)
- [ ] not applicable
- [ ] other (enter in the "Additional environment details" area below)

### Additional environment details

- browser version
- SPFx version
- Node.js version
- etc

### Describe the bug / error

After follow guideline from @lucabandMSFT My Sharepoint Admin Access token has been broken.
https://github.com/SharePoint/sp-dev-docs/issues/7640#issuecomment-1021519426
![image](https://user-images.githubusercontent.com/97509402/153826844-b51442ad-393d-48d6-a6b8-74cd28790b7f.png)

### Steps to reproduce

[](https://user-images.githubusercontent.com/46542683/151042941-8cc138be-f0c6-4395-84c6-36c403d37d22.png)
log in to Azure portal as a Global Administrator
go to "App registrations"
search for "SharePoint Online Client Extensibility Web Application Principal" object
click there and select "Certificates and secrets"; you should end up in a UX like the following and you should have just one client secret
image
at that point click on the trash can icon and delete that secret. Once you are done there. go back to SPO tenant admin and ensure you are both Global Admin AND SPO teantn admin. Then go to the "API Access" page: that will trigger generation of a new client secret (you can see that in the AAD portal after you visited the page in SPO tenant admin and refresh).

### Expected behavior

It should generate new secret and spfx app work as expected on Team Desktop client.

**Comment by ntdong1889** [comment]
Here is my log of the page
[ApiAccessLog.zip](https://github.com/SharePoint/sp-dev-docs/files/8059095/ApiAccessLog.zip)

**Comment by lucabandMSFT** [comment]
thanks @ntdong1889. looking at that logs now.

**Comment by lucabandMSFT** [comment]
hey @ntdong1889 , can you please go to the AAD Portal and for that principal tell me what the identifierUris is? you can find it under "manifest" section.

**Comment by GrahamMcMynn** [comment]
@ntdong1889  - Is this actually continuing to fail?  I noticed you sent me a trace from January 27th.

The issue you are seeing looks like it is because the identifier uri already exists... however, if it already exists we shouldn't try to create a new one.  This can happen if the service principal was just created and we query for it, so we don't find it (due to propagation being slow), then when we try to create it you get an error that it already exists.  However, when that happens refreshing the page fixes that as it has now had time to propagate.  Is it still occuring?

**Comment by ntdong1889** [comment]
> hey @ntdong1889 , can you please go to the AAD Portal and for that principal tell me what the identifierUris is? you can find it under "manifest" section.

![image](https://user-images.githubusercontent.com/97509402/153985664-e9cb82eb-4d2f-4f13-ab73-00b5e7c4979f.png)

**Comment by ntdong1889** [comment]
> @ntdong1889 - Is this actually continuing to fail? I noticed you sent me a trace from January 27th.
> 
> The issue you are seeing looks like it is because the identifier uri already exists... however, if it already exists we shouldn't try to create a new one. This can happen if the service principal was just created and we query for it, so we don't find it (due to propagation being slow), then when we try to create it you get an error that it already exists. However, when that happens refreshing the page fixes that as it has now had time to propagate. Is it still occuring?

Yes it's a continuing to fail. I ask all the SP admin to try but same result. :(

**Comment by ntdong1889** [comment]
@GrahamMcMynn @lucabandMSFT  Do we have any news on this please :). Thanks for all the support

**Comment by ntdong1889** [comment]
> Hi @ntdong1889 - sorry for missing your original response. The issue is the identifier is not updated (as can be seen by the image you previously sent) . You need a global admin to visit the web api access page (not just a SP admin). If you are visiting as a global admin please send a trace of that to me so I can take a look at it if it continues not to work.

Thank I will contact my boss and use the highest level account to check this out. I will let you know about this.
Thank for all the support.

**Comment by ntdong1889** [comment]
> Hi @ntdong1889 - sorry for missing your original response. The issue is the identifier is not updated (as can be seen by the image you previously sent) . You need a global admin to visit the web api access page (not just a SP admin). If you are visiting as a global admin please send a trace of that to me so I can take a look at it if it continues not to work.

Hi I just check with Top account and still the same issue.

**Comment by GrahamMcMynn** [comment]
HI @ntdong1889  - I understand your concern.  Has the identifier URI been updated?  Please go to the AAD Portal and for that principal tell if it has changed.  You can find it under "manifest" section.

If it still continues to be https://microsoft.spfx3rdparty.com that is still the issue.  If that is the case can you at a minimum have your boss visit the api access page and tell me the UTC time that he does that at.  I can pull logs from my side, but I need that information.

**Comment by ntdong1889** [comment]
> HI @ntdong1889 - I understand your concern. Has the identifier URI been updated? Please go to the AAD Portal and for that principal tell if it has changed. You can find it under "manifest" section.
> 
> If it still continues to be https://microsoft.spfx3rdparty.com that is still the issue. If that is the case can you at a minimum have your boss visit the api access page and tell me the UTC time that he does that at. I can pull logs from my side, but I need that information.

Hi @GrahamMcMynn ,

It is still https://microsoft.spfx3rdparty.com/ like the picture.
![image](https://user-images.githubusercontent.com/97509402/157366736-1b6b1ead-3313-492a-b330-c4be29a09dee.png)
My Boss Access around 08-03-2022 2:30pm to 3:00pm GMT+7 (7:30am-8:00amUTC)
![image](https://user-images.githubusercontent.com/97509402/157366787-b6449d03-1ffc-42f5-a139-8ad0840d7b9e.png)

Thank you alot for all the support 👍  💯 💯 💯 💯

**Comment by GrahamMcMynn** [comment]
@ntdong1889 - Thank you for sending that info.  I was able to look up in our logs and see what is going on.  Our code is failing to set the identifierUri (I guess that could have been expected seeming as it has the old value).

It's getting the following error:
Unexpected failure while adding resource identifier. Error: Property identifierUris is invalid.

That happens when trying to set the identifierUris to be:
"identifierUris": [
"api://ce57ebe3-a63d-4708-b5cf-c274b48bd26c/microsoft.spfx3rdparty.com
]

I am looking into it from our side why that won't update... however, you will be able to unblock yourself if you manually modify that identifier URI.

Thanks,
Graham

**Comment by GrahamMcMynn** [comment]
@ntdong1889  - could you do the following:
1.  Visit https://developer.microsoft.com/en-us/graph/graph-explorer
2. Run the following get query:
  https://graph.microsoft.com/v1.0/applications?$filter=identifierUris/any(r:r+eq+ 'api://ce57ebe3-a63d-4708-b5cf-c274b48bd26c/microsoft.spfx3rdparty.com')

If there are results copy the results and carry on, if there are no results please let me know.

3.  Run the following get query:  
    https://graph.microsoft.com/v1.0/applications?$filter=identifierUris/any(r:r+eq+ 'https://microsoft.spfx3rdparty.com')

See that those values are different.  If they are different the issue is that both exist.

**Comment by ntdong1889** [comment]
The first query return empty.
![image](https://user-images.githubusercontent.com/97509402/157580518-5df53401-f527-4727-9265-8a193a0a7c7e.png)
The second return value like picture 
![image](https://user-images.githubusercontent.com/97509402/157580664-db12558c-f52a-40cf-a937-b0d35273e669.png)

**Comment by GrahamMcMynn** [comment]
Hi @ntdong1889 - Ok that is not what I expected.  Would it be possible for you to update the identifier URI manually to the value I gave via portal.azure.com and if that fails send a trace for that?

**Comment by GrahamMcMynn** [comment]
@ntdong1889 - yes, update it as suggested.  If you hit an error updating would it be possible to send a trace?  If so please send it to grahamc at microsoft.com.

**Comment by ntdong1889** [comment]
> @ntdong1889 - yes, update it as suggested. If you hit an error updating would it be possible to send a trace? If so please send it to grahamc at microsoft.com.

I successes update to the new value "api://ce57ebe3-a63d-4708-b5cf-c274b48bd26c/microsoft.spfx3rdparty.com"
but this part still not work
> at that point click on the trash can icon and delete that secret. Once you are done there. go back to SPO tenant admin and ensure you are both Global Admin AND SPO teantn admin. Then go to the "API Access" page: that will trigger generation of a new client secret (you can see that in the AAD portal after you visited the page in SPO tenant admin and refresh).

**Comment by ntdong1889** [comment]
> @ntdong1889 - I'm afraid I don't understand. Why are you deleting the service principal? I actually need more information if it still doesn't work... could you send me a time when you tried it, a correlation id, or even better a trace and description of what didn't work? I'm sorry you're running into this problem and I would really like to help you and figure out what is wrong so we can fix it for everyone affected by this issue.

After doing it All ours app use graph API stop working even on web browser. And I can not revert it back to previous value 
![image](https://user-images.githubusercontent.com/97509402/159215553-c4de9597-bc3a-4447-a513-8a2f0287d077.png)

**Comment by ntdong1889** [comment]
@GrahamMcMynn Finally I have a solution that We delete the hold app SharePoint Online Client Extensibility Web Application Principal and SharePoint Online Client Extensibility Web Application Principal Helper and let the AAD create them again. After that request permission from app and go to API access page to grand again. Now all my app work as expected.

Thank for all your help this idea could help other people to solve the Custom SPFX app not work on Teams Clients
